packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -733,6 +733,40 @@ const PublishAction: DocumentActionComponent = ({
 
 PublishAction.type = 'publish';
 
+const formDocumentToData = (document: any) => {
+  return Object.keys(document).reduce((acc, key) => {
```

Comment:
Don't you need to apply the logic for relations inside components too? You may need to traverse the document tree. There's traverseAndExtract in this file that does something similar

Reply:
Yes definitly needed for components